### ISSUE TITLE
barcodes

### ISSUE THREAD

**Issue by byuherpetology** [ArctosDB/data-migration#108]
I am trying to add barcodes for tissue samples that have previously had no barcode.  It seems the easiest way is to add the barcodes to the bulkload file, then go back and apply the appropriate barcode to the tissue sample.  I guess I would also need to assign containers and locations after the data has been bulkloaded.  

I know barcode #'s need to be assigned by Arctos, how do I go about getting a set of barcodes designated for our collection?

**Comment by Jegelewicz** [comment]
Start be reading the [Object Tracking Documentation](http://handbook.arctosdb.org/documentation/container.html) and [How To Start Object Tracking in Arctos](http://handbook.arctosdb.org/how_to/How-to-Start-Object-Tracking.html). When you are ready to claim a barcode series, let us know and we can help. Also just post questions here!

**Comment by byuherpetology** [comment]
We are ready for a series of 10,000 barcodes.  We also need to have access to locations so we can create and edit containers.

**Comment by Jegelewicz** [comment]
The first step is to decide what you want your series to be. See http://handbook.arctosdb.org/how_to/How-to-Start-Object-Tracking.html#develop-your-barcode-series

**Comment by byuherpetology** [comment]
Ok, I would like to claim the series BYUGC 10000-BYUGC 99999.  I do not have the manage container permission, so I can't check to see if this series is in use - but I doubt it.  

For the cryovials, do I design the series of random letter and number barcodes for us to claim?  Or is this series randomly generated and provided to us?

**Comment by Jegelewicz** [comment]
> BYUGC 10000-BYUGC 99999

Spaces are not a good thing. How about
BYUGC10000-BYUGC99999

> do I design the series of random letter and number barcodes for us to claim? Or is this series randomly generated and provided to us?

You design them. Let me know if you need any assistance. @campmlc probably has some good advice!

**Comment by dustymc** [comment]
Yes agreed - that space (or was it one of the dozens of unicode characters that sorta look like a space in various UIs?!?) is likely to make various devices angry, and it will limit what can be done with those barcodes. See http://handbook.arctosdb.org/documentation/container.html#guidelines-for-barcode-containing-labels.

I suppose "random" is possible, but I certainly wouldn't want to manage it or try to figure out what the second batch might look like! Base36 (or 64, or whatever) encoding numeric values probably looks random to most users but comes with none of the problems a random "series" would.

**Comment by byuherpetology** [comment]
Got it, no space.  I'm sure this was just my inattention to detail.  so I think we are set with the series for large containers. 
On the cryovial barcodes, in the documentation it says "Small containers (cryovials) have a randomly generated combination of capital letters A-Z and numbers 0-9 in the range “A plus any four upper-letters or numbers”.  
I think the "randomly generated" threw me.  So what we need is a series with a set prefix and a set number of digits/letters?  Is it unwise to use the BYUXXXX?  is that too close to our container series?

**Comment by Jegelewicz** [comment]
> I think the "randomly generated" threw me. So what we need is a series with a set prefix and a set number of digits/letters? Is it unwise to use BYUXXXX? is that too close to our container series?

BYU is too close to your catalog numbers. At the Museum of Southwestern Bio, they use a base 36 series that always begins with the letter A.

> A24E3 - AZZZZ reserved May 2016; = "A plus any four upper-letters or numbers"

How about this system but starting with a B?

**Comment by Jegelewicz** [comment]
For your long barcodes, I claimed the following:

> BYUGC00000-BYUGC99999 claimed February 4, 2020

This way you could also access BYUGC00000 through BYUGC09999 if you need them in the future.

**Comment by Jegelewicz** [comment]
Also, before I give you access to object tracking, please read and acknowledge that you understand the following:

[Arctos Container Documentation](http://handbook.arctosdb.org/documentation/container.html)

[How to Start Object Tracking in Arctos](http://handbook.arctosdb.org/how_to/How-to-Start-Object-Tracking.html)

**Comment by byuherpetology** [comment]
> BYU is too close to your catalog numbers. At the Museum of Southwestern Bio, they use a base 36 series that always begins with the letter A.

> A24E3 - AZZZZ reserved May 2016; = "A plus any four upper-letters or numbers"

> How about this system but starting with a B?

Very true, BYUXXXX would be a mess with catalog numbers.  I am happy with B plus four upper-letters or numbers.  

> Also, before I give you access to object tracking, please read and acknowledge that you understand the following:

> Arctos Container Documentation

> How to Start Object Tracking in Arctos

I have read both of these pages multiple times and I think I understand them, but as you can see from my continued questions, that is not always the case.  I will be cautious and ask before doing anything I am unsure of.  
Thank you, you are super helpful as always!

**Comment by byuherpetology** [comment]
We have a new employee who will be in charge of the genomic resources.  I would like to get her on github with access to Arctos DB.  Her user name is kokkoana27.

**Comment by Jegelewicz** [comment]
Yay! Sent her an invite to join the team.

**Comment by Jegelewicz** [comment]
Also granted @byuherpetology access to object tracking.

**Comment by Jegelewicz** [comment]
And finally, reserved barcodes as follows:

B0000 - BZZZZ "B plus any four upper-letters or numbers"
